docs/index.rst

```diff
@@ -103,15 +107,15 @@ The Python bindings also provide some additional configuration properties:
   (or ``on_delivery=callable``) to the confluent_kafka.Producer.produce() function.
   Currently message headers are not supported on the message returned to the
   callback. The ``msg.headers()`` will return None even if the original message
-  had headers set.
+  had headers set. This callback is served upon calling ``client.poll()`` or ``producer.flush()``.
```

Comment:
We can be more explicit and say producer.poll here since it is only valid for the producer